.github/workflows/e2e.yml

```diff
@@ -81,16 +133,53 @@ jobs:
           DOCKERFILE_PATH: './docker/Dockerfile'
         run: |
           docker build \
-          -t "${{ env.DATA_IMAGE_TAG }}" \
+          -t "${{ env.DATA_IMAGE_PATH }}" \
           -f "${{ env.DOCKERFILE_PATH }}" \
+          --build-arg SOURCE_REPO="https://github.com/${{ env.NAMESPACE }}/${{ vars.DB_NAME }}" \
+          --build-arg DESCRIPTION="${GITHUB_SHA::7}" \
           --build-arg LOCAL_DATA_PATH="${{ env.LOCAL_DATA_PATH }}" \
           --build-arg LOCAL_COLLECTIONS_PATH="${{ env.LOCAL_COLLECTIONS_PATH }}" \
           --build-arg LOCAL_IMPORT_SCRIPT_PATH="${{ env.LOCAL_IMPORT_SCRIPT_PATH }}" \
           --build-arg LOCAL_CREATE_INDEXES_PATH="${{ env.LOCAL_CREATE_INDEXES_PATH }}" \
           .
-      # TODO - push image to remote repository
 
-      # TODO - move into job that pulls image from remote repository
+      - name: login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Note: if a new package is created using this workflow, then the package
+      # visibility will default to the source repository visbility. For example,
+      # a package created by a workflow in a public repo will also be public.
+      # Publicly visible container packages are anonymously downloadable.
+      - name: push to GitHub Container Registry
+        run: docker push "${{ env.DATA_IMAGE_PATH }}"
+
+  run-e2e-tests:
+    needs:
+      - check-for-image
+      - build-image
+    # always() prevents skipping this job when build-image was skipped
+    # see https://github.com/actions/runner/issues/491
+    if: ${{ always() && needs.check-for-image.result == 'success' && (needs.build-image.result == 'success' || needs.build-image.result == 'skipped')}}
```

Comment:
The workflow will run the run-e2e-test job even if the image did not need to be built, as in [this run](https://github.com/hallieswan/Agora/actions/runs/9508204995/attempts/1).